hrjobcontract/js/src/job-contract/controllers/form/form-leave.js

```diff
@@ -0,0 +1,38 @@
+define([
+    'job-contract/vendor/fraction',
+    'job-contract/controllers/controllers'
+], function (Fraction, controllers) {
+    'use strict';
+
+    controllers.controller('FormLeaveCtrl',['$scope', '$rootScope', '$filter', '$log', '$q', 'UtilsService',
+      function ($scope, $rootScope, $filter, $log, $q, UtilsService) {
+        $log.debug('Controller: FormLeaveCtrl');
+
+        var promiseNumberOfPublicHolidays = UtilsService.getNumberOfPublicHolidaysInCurrentPeriod();
+
+        promiseNumberOfPublicHolidays.then(function(result){
+          var number = parseFloat(result);
+          if(isNaN(number)) {
+            number = 0;
+          }
+          $scope.numberOfPublicHolidays = number;
+        });
+
+        $scope.$watch('entity.leave', function(newValue, oldValue) {
+          var newTypeWithPublicHolidays = null;
+          newValue.forEach(function(value, i) {
+            if(value.add_public_holidays === '1' && oldValue[i].add_public_holidays === '0') {
```

Comment:
Why are you not simply using `true` and `false` here? If it's because the DB is storing this as a 1/0 value, I'm not 100% sure but i think you can still pass a bool value to the API and it will be correctly converted to the expected value anyway. This could solve a lot of this ugliness with `'1'` and `'0'` that i see going on here